.github/workflows/main.yml

```diff
@@ -65,15 +65,24 @@ jobs:
         with:
           version: ${{ needs.determineVersionNumber.outputs.version }}
 
-      - name: Upload NuGetForUnity.dll
+      - name: Upload NuGetForUnity dlls
         uses: actions/upload-artifact@v3
         with:
-          name: NuGetForUnity.dll
-          path: ./src/NuGetForUnity.CreateDll/bin/Release/NugetForUnity.dll
+          name: NuGetForUnity-dlls
+          path: |
+            ./src/NuGetForUnity.CreateDll/bin/Release/NuGetForUnity.dll
+            ./src/NuGetForUnity.CreateDll/bin/Release/NuGetForUnity.PluginAPI.dll
+            ./src/NuGetForUnity.CreateDll/bin/Release/NuGetForUnity.PluginAPI.xml
           if-no-files-found: error
 
+      - name: Ensure NuGetForUnity.PluginAPI.dll in GIT is up-to-date (if it fails pleas build the NugetForUnity.PluginAPI.csproj and upload the changed files to git)
+        run: git diff --exit-code ./src/NuGetForUnity/Editor/PluginAPI/NuGetForUnity.PluginAPI.dll
+
+      - name: Ensure NuGetForUnity.PluginAPI.xml in GIT is up-to-date (if it fails pleas build the NugetForUnity.PluginAPI.csproj and upload the changed files to git)
+        run: git diff --exit-code ./src/NuGetForUnity/Editor/PluginAPI/NuGetForUnity.PluginAPI.xml
```

Comment:
What is the intent here? Will it check that each commit has changes in these files? They should only be updated if something in PluginAPI project was changed which should be pretty rare.

Reply:
No they should only check if `NuGetForUnity.PluginAPI.dll` is updated if something it the code of the PluginAPI project was changed. It currenlty has the issue that the `.dll` contains the version number so it always failes.